World.py

```diff
@@ -78,7 +78,16 @@ def __init__(self, world_id: int, settings: Settings, resolve_randomized_setting
         self.disable_trade_revert: bool = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or settings.adult_trade_shuffle
         self.skip_child_zelda: bool = 'Zeldas Letter' not in settings.shuffle_child_trade and \
                                       'Zeldas Letter' in self.distribution.starting_items
-        self.selected_adult_trade_item: str = ''
+        self.selected_adult_trade_item: str = random.choice(settings.adult_trade_start) if settings.adult_trade_start else None
+        # Override the adult trade item used to control trade quest flags during patching if any are placed in plando.
+        # This has to run here because the rule parser caches world attributes and this attribute impacts logic for buying a blue potion from Granny's Potion shop.
+        locations = {}
+        if self.distribution.locations:
+            locations = {loc: self.distribution.locations[loc] for loc in random.sample(sorted(self.distribution.locations), len(self.distribution.locations))}
```

Comment:
This line isn't necessary, it just shuffles locations around to fill in a random order. Using `self.distribution.locations` directly works fine in the filter.